### ISSUE TITLE
Support hashes ..

### ISSUE THREAD

**Issue by skx** [skx/evalfilter#154]
Hashes are an obvious data-structure to add, and the changes won't be too hard to add:

* Hashes can be keyed by "string" or "number".
* Hashes can be declared literally.
* Hashes might be returned by some built-ins.

Changes to be made, in brief:

* [x] The parser needs to recognize a hash literal, at least.
* [x] We need a Hash-object.
* [x] The run-time needs to be updated to handle hash-objects.  e.g. `foreach`.
* [x] The run-time built-ins such as `len`, need update.
* [x] We need a new "keys" built-in - or defer to `foreach`.
* [x] The VM needs an `OpHash` to initialize a hash, at least.

Test-case coverage should be high.  We're in good shape there generally.

**Comment by skx** [comment]
Test program:

```
$ cat hash.in
a = { "Name": "Steve", Age: 2020 - 1976 }

printf("I am %s - %d\n", a["Name"], a["Age"] );
```

**Comment by skx** [comment]
Test program works; more work to be done still to add test-cases, update the builtins, and etc.

**Comment by skx** [comment]
Closed in #155
